### DISCUSSION TITLE
Jump Vehicle If There is any block in front

### DISCUSSION THREAD

**Discussion by Codestech1** [Minestom/Minestom#1611] [Help]
Hi, i wanna add to my mount system a way to auto jump blocks when there are any block in front. Can you help me?
Code:
```
listener.addListener(PlayerTickEvent.class, event -> {
            PlayerVehicleInformation info = getOwner().getPlayer().getVehicleInformation();
            Pos pos = getOwner().getPlayer().getPosition();
            if(info.shouldUnmount()) {
                clear();
                return;
            }
            Vec vec = pos.direction().withY(0);
            mount.setView(pos.yaw(), pos.pitch());
            if(info.getForward() > 0) {
                vec = vec.mul(getOwner().getPlayer().getAttribute(Attribute.MOVEMENT_SPEED).getValue() * 60);
                mount.setVelocity(vec);
            } else if(info.getForward() < 0) {
                vec = vec.mul((getOwner().getPlayer().getAttribute(Attribute.MOVEMENT_SPEED).getValue() * -1) * 60);
                mount.setVelocity(vec);
            }
        });
```
![image](https://user-images.githubusercontent.com/56263050/209167845-0ddd26eb-50aa-4927-b190-97823682e911.png)

**Comment by yakovliam** [comment]
Here's an example that is implemented using my 3D physics engine, so all you should need to do is develop a method for detecting if the horse is colliding with any blocks.

https://github.com/skycade/skycade-tanks/blob/862eea4b7505d138ce458eb1222584c14a054b15/src/main/java/net/skycade/tanks/physics/tank/TankObject.java#L78
